### DISCUSSION TITLE
How to import recipes from files (i.e. HTML)?

### DISCUSSION THREAD

**Discussion by ovizii** [mealie-recipes/mealie#3644] [Q&A] [answered]
Let me explain:

Sometimes, I find a recipe published without recipe markup so mealie cannot import it. 

I found that tasking ChatGPT with this prompt works wonders. 

Please scrape this site https://site.tld/recipe-example for the recipe and give me the scraped recipe formatted with recipe markup as a HTML file for download.

Now I am just wondering about the easiest way to import it into mealie. I don't really want to fire up a local web server to serve the file so mealie can scrape it.

Any ideas?

**Comment by ovizii** [comment]
Just to keep this discussion updated, I found partial solutions here: https://www.reddit.com/r/selfhosted/comments/11ptvpp/python_script_that_imports_recipes_from_a_url/ and here: https://github.com/mealie-recipes/mealie/discussions/2264 

Still playing around with those ideas.

**Comment by michael-genson** [comment, marked as answer]
FYI, the recipe scraper now falls back to OpenAI if it fails to parse normally: https://github.com/mealie-recipes/mealie/pull/3690

The next step is to enable this for uploaded files, too

**Reply by ovizii** [reply]
Oh, wow! I have to check what version I am on and what version this is merged into, but that is great news.

**Comment by ovizii** [comment]
**Guess I'm not there yet. OpenAI is enabled and works for ingredients parsing.** 

![image](https://github.com/mealie-recipes/mealie/assets/837902/a7ae943b-c0b3-49b2-9b9a-f3403359e221)

![image](https://github.com/mealie-recipes/mealie/assets/837902/128b50a4-df4d-4124-9fbb-fb33128890ed)

**Reply by michael-genson** [reply]
It's currently only on nightly, it will be in the next release

**Reply by ovizii** [reply]
Works very nicely :-)

**Reply by ovizii** [reply]
I replied too soon. My second try with a page without recipe markup mealie failed and just said:

![image](https://github.com/mealie-recipes/mealie/assets/837902/6f73d724-31ae-45b5-83e6-841b68db7aa2)

URL: https://ottolenghi.co.uk/recipes/pork-souvlaki-with-cucumber-salad-and-tzatziki

Prompting ChatGPT directly works:

> Can you scrape the recipe from this site and offer it to me in HTML format with the proper recipe markup for download? https://ottolenghi.co.uk/recipes/pork-souvlaki-with-cucumber-salad-and-tzatziki

**Reply by michael-genson** [reply]
Ah yep that's a bug

**Reply by michael-genson** [reply]
See: https://github.com/mealie-recipes/mealie/pull/3749

**Reply by ovizii** [reply]
Not sure its the same problem as I don't see any errors except this one in the docker logs:

` INFO     2024-06-22T21:56:02 - HTTP Request: GET https://www.thekitchn.com/arayes-lebanese-stuffed-pita-recipe-23664768 "HTTP/1.1 403 Forbidden"`

Here is another URL which gives the exact same error as in my last screenshot. https://www.thekitchn.com/arayes-lebanese-stuffed-pita-recipe-23664768

**Reply by michael-genson** [reply]
The 403 error is unrelated to Mealie; this means the website is blocking scraping traffic (the website is returning 403, not Mealie). If we can't fetch the content, we can't do anything with it
